server/matchmaker/search.py

```diff
@@ -40,9 +43,29 @@ def __init__(self, player, start_time=None, rating_prop='ladder_rating'):
             0: 0.8
         }
 
+        self.adjusted_rating = self.adjust_rating()
+
+    def adjust_rating(self):
+        if self.rating_prop=='ladder_rating':
+            numgames = self.player.numGames
+            if numgames <= self.noobFixMaxGames:
+                mean, dev = self.player.ladder_rating
+                # A simple linear interpolation
+                adjusted_mean = ((self.noobFixMaxGames - numgames) * self.noobFixBaseRating + numgames * mean) / self.noobFixMaxGames
+                self._logger.info('Adjusted mean rating for {player} with {numgames} games from {mean} to {adjusted_mean}'.format(
```

Comment:
This logging statement will be very noisy -- it would be preferable to put it elsewhere, e.g. where the match is actually made here: https://github.com/FAForever/server/blob/develop/server/matchmaker/search.py#L125